addons/purchase/models/purchase.py

```diff
@@ -109,7 +97,7 @@ def _compute_invoice(self):
     ], string='Billing Status', compute='_get_invoiced', store=True, readonly=True, copy=False, default='no')
 
     # There is no inverse function on purpose since the date may be different on each line
-    date_planned = fields.Datetime(string='Scheduled Date', compute='_compute_date_planned', store=True, index=True)
+    date_planned = fields.Datetime(string='Scheduled Date', index=True)
```

Comment:
Just a comment: (Not sure If it fulfill whole requirement)
Can't we use stored related field here instead of doing similar things on create/write ?